server/pulp/server/db/connection.py

```diff
@@ -91,12 +98,26 @@ def initialize(name=None, seeds=None, max_pool_size=None, replica_set=None, max_
                                                                 itertools.repeat(32))
 
         if seeds != '':
-            seeds_list = seeds.split(',')
             if len(seeds_list) > 1 and not replica_set:
                 raise PulpCodedException(error_code=error_codes.PLP0041)
             while True:
                 _CONNECTION = _connect_to_one_of_seeds(connection_kwargs, seeds_list, name)
                 if _CONNECTION:
+                    db_min_version_tuple = tuple(MONGO_MINIMUM_VERSION.split("."))
+                    wc_min_version_tuple = tuple(MONGO_WRITE_CONCERN_VERSION.split("."))
+                    db_version_tuple = _get_mongo_version()
+                    if db_version_tuple < db_min_version_tuple:
+                        raise RuntimeError(_("Pulp requires Mongo version %s, but DB is reporting"
+                                             "version %s") % (MONGO_MINIMUM_VERSION,
+                                                              '.'.join(db_version_tuple)))
+                    elif db_version_tuple >= wc_min_version_tuple or replica_set:
+                        # Write concern of 'majority' only works with a replica set or when using
+                        # MongoDB >= 2.6.0
+                        _CONNECTION.write_concern['w'] = write_concern
```

Comment:
It seems that if a user sets write concern while using mongo 2.4, we should raise an error. This code will set the write concern to 1 instead.